Sources/SwifterSwift/UIKit/UIButtonExtensions.swift

```diff
@@ -204,7 +204,28 @@ public extension UIButton {
             contentEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: insetAmount)
         }
     }
-
+    
+    /// SwifterSwift: Put image to the right side of title
+    /// 
+    /// - Parameter spacing: spacing between title text and image.
+    func rightAlignImage(spacing: CGFloat) {
+        guard
+            let imageSize = imageView?.image?.size,
+            let text = titleLabel?.text,
+            let font = titleLabel?.font
+        else { return }
+        
+        let titleSize = text.size(withAttributes: [.font: font])
```

Comment:
There is similar code to [#807](https://github.com/SwifterSwift/SwifterSwift/pull/807). I think it could be combined into one method, but in order to avoid creating a new type (enum like `ImagePosition` in [a commit](https://github.com/SwifterSwift/SwifterSwift/pull/807/commits/bba4625a6c1330942e6c4a2fb1aa82bc07974786) of 807), Maybe you can implement it with two parameters, similar to `UIStackView`'s axis and alignment. such as (you should find a better name, my English is not good, :-)):

```swift
func centerImageAndText(axis: NSLayoutConstraint.Axis, imageAlignment: UIStackView.Alignment, spacing: CGFloat)
```

Reply:
+1

Reply:
Hi @qchenqizhi Thank you for your comments. But, do you mean I will have to stop adding my simple method `rightAlignImage(spacing:)`, instead, creating another method to handle all the different positioning of `image` and `label` for `UIButton`?

In summary, these are the reasons why I didn't mix `rightAlignImge(spacing:)` with your `centerImageAndText` method below:

1. personally, I think the extension method should be simple and effective.
2. positioning image to the left of label on UIButton is some default behaviour, do we need to override it?

Reply:
You are welcome. So we will have two methods:
one supports the image on the left of text, or the image above text,
and the other supports the image on the right of text.
Isn't this strange?

Reply:
Why is it strange? 
It sounds to me that you are insisting I should create one big method to covering all the cases? I think that is strange. @qchenqizhi

Reply:
It seems a deadlock for me to create this PR, however, I feel like to contribute to this project. Is there anyone else could help with this situation? Much appreciated!

Reply:
Honestly, it's okay, but I don't think the code is very tidy. You guys decide. @LucianoPAlmeida @guykogus

Reply:
I think it's fine to have this method separate to the other one.
Personally I wouldn't use any of these anyway as they assume that the button's frame will never change. I have my own custom `UIButton` subclasses that override `titleRect(forContentRect:)` and `imageRect(forContentRect)`.